iceoryx_posh/test/moduletests/test_popo_waitset.cpp

```diff
@@ -121,6 +126,31 @@ class WaitSet_test : public Test
 
     void TearDown(){};
 
+    template <uint64_t EventInfoVectorCapacity, typename EventOrigin>
+    static bool
+    doesEventInfoVectorContain(const iox::cxx::vector<const EventInfo*, EventInfoVectorCapacity>& eventInfoVector,
+                               const uint64_t eventId,
+                               const EventOrigin& origin)
+    {
+        for (auto& e : eventInfoVector)
+        {
+            if (e->getEventId() == eventId && e->doesOriginateFrom(&origin)
+                && e->template getOrigin<EventOrigin>() == &origin)
```

Comment:
I didn't look at the code but isn't this line superfluous?

Reply:
Actually no, it could be possible that the type tracking is wrong and then getOrigin calls the errorHandler since you would like to acquire the pointer of the wrong type.